fe/fe-core/src/main/java/com/starrocks/qe/scheduler/DefaultWorkerProvider.java

```diff
@@ -326,8 +332,26 @@ private static ImmutableMap<Long, ComputeNode> buildComputeNodeInfo(SystemInfoSe
                 }
                 computeNodes.put(computeNode.getId(), computeNode);
             }
-            return ImmutableMap.copyOf(computeNodes);
         }
+
+        //add BE to Node Pool
+        if (computationFragmentSchedulingPolicy.equals("all_nodes")) {
+            ImmutableMap<Long, ComputeNode> idToBackend
+                    = ImmutableMap.copyOf(systemInfoService.getIdToBackend());
+            for (int i = 0; i < idToBackend.size(); i++) {
+                ComputeNode backend =
```

Comment:
does the total number of compute nodes will still be enforced by taking the be as computenodes?

e.g. if the system has 2 cn and 3 be nodes and `numUsedComputeNodes` is set to `3`, it will be 2 cn nodes with one additional be node, or it will be 2 cn nodes plus all 3 be nodes?

Reply:
It should be 2 cn nodes plus all 3 be nodes. when face sudden business peak, current be have not enough computational power, need deploy additional cn to increace. Then total computational power should be all be and numUsedComputeNodes's cn.

Reply:
I don't think so.

This new session variable just controls whether need to take BE nodes as CN nodes. whether to use cn nodes, how many cn nodes to be used, will be still controlled by `preferComputeNode` and `useComputeNodes`, you can't say if use `all_nodes`, even the session variable set `preferComputeNode=false`, the compute nodes will be used anyway, that's doesn't make sense.

So in normal case
* `preferComputeNode = true`
* `useComputeNodes = -1`
* `computation_fragment_scheduling_policy = compute_nodes_only `

This will use only compute nodes all the computation works, BE will be excluded.

And when facing the sudden peak load, set the session variable
* `computation_fragment_scheduling_policy = all_nodes ` to let BE nodes working as compute nodes as well.

`preferComputeNode` and `useComputeNodes` should be always obeyed no matter what `computation_fragment_scheduling_policy` is.

Reply:
yes , the pr is obey the policy as you say. I offer the FE UT in DefaultWorkerProviderTest.java, the testSelectBackendAndComputeNode function display the test demo:
SystemInfoService has 5 Backend and 3 ComputeNode
- case 1.test ComputeNode only：preferComputeNode=true、useComputeNodes=[-1, 0, 2, 4]、computation_fragment_scheduling_policy=compute_nodes_only
-  case 2.test Backend only：preferComputeNode=false、useComputeNodes=[-1, 0, 2, 4]、computation_fragment_scheduling_policy=compute_nodes_only
-  case 3.test Backend and ComputeNode：preferComputeNode=true、useComputeNodes=[-1, 0, 2, 4]、computation_fragment_scheduling_policy=all_nodes